FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,368 @@
+		max_jobs=$max_jobs
+    if (( max_jobs < 1 )); then max_jobs=2; fi 
+    ;;
+	NONE)
+    echo "parallel mode: NONE" 
+    ;;
+	*)
+		log_Err_Abort "Unknown parallel mode $parallel_mode. Plese specify one of FSLSUB, BUILTIN, NONE"
+		;;
+esac
+
+start_stage=0
+if [ -n "$StartStage" ]; then	
+	case $StartStage in
+    TEMPLATE) start_stage=0 ;;
+		TIMEPOINTS) start_stage=1 ;;
+		*) log_Err_Abort "Unrecognized option for start-stage: $StartStage"
+	esac
+fi
+end_stage=1
+if [ -n "$EndStage" ]; then	
+	case $EndStage in
+		TEMPLATE) end_stage=0 ;;
+		TIMEPOINTS) end_stage=1 ;;
+		*) log_Err_Abort "Unrecognized option for end-stage: $EndStage"
+	esac
+fi
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+# ----------------------------------------------------------------------
+# Log values retrieved from positional parameters
+# ----------------------------------------------------------------------
+log_Msg "StudyFolder: ${StudyFolder}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "Sessions: ${Sessions}"
+log_Msg "TemplateID: ${TemplateID}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "StartStage: $StartStage"
+log_Msg "EndStage: $EndStage"
+
+# ----------------------------------------------------------------------
+log_Msg "Preparing the folder structure"
+# ----------------------------------------------------------------------
+Sessions=`echo ${Sessions} | sed 's/,/ /g'`
+
+extra_reconall_args_base=""
+extra_reconall_args_long=""
+
+Session0=( $Sessions ); Session0=${Session0[0]}
+if (( UseT2w )); then
+  extra_reconall_args_base="-T2pial -T2 ${StudyFolder}/${Session0}/T1w/T2w_acpc_dc_restore.nii.gz"
+  extra_reconall_args_long="-T2pial"
+fi
+
+log_Msg "extra_reconall_args_base: $extra_reconall_args_base"
+log_Msg "extra_reconall_args_long: $extra_reconall_args_long"
+log_Msg "After delimiter substitution, Sessions: ${Sessions}"
+
+LongDIR="${StudyFolder}/${SubjectID}.long.${TemplateID}/T1w"
+mkdir -p "${LongDIR}"
+
+for Session in ${Sessions} ; do
+  Source="${StudyFolder}/${Session}/T1w/${Session}"
+  Target="${LongDIR}/${Session}"
+  log_Msg "Creating a link: ${Source} => ${Target}"
+  ln -sf ${Source} ${Target}
+done
+
+
+if (( start_stage < 1 )); then 
+
+	# ----------------------------------------------------------------------
+  log_Msg "Creating the base template: ${TemplateID}"
+	# ----------------------------------------------------------------------
+	# backup template dir if it exists
+  if [ -d "${LongDIR}/${TemplateID}" ]; then
+    TimeStamp=`date +%Y-%m-%d_%H.%M.%S.%6N`
+	  log_Msg "Base template dir: ${LongDIR}/${TemplateID} already exists, backing up to ${LongDIR}/${TemplateID}.${TimeStamp}"
+	  mv ${LongDIR}/${TemplateID} ${LongDIR}/${TemplateID}.${TimeStamp}
+	fi
+
+  recon_all_cmd="recon-all.v6.hires"
+  recon_all_cmd+=" -sd ${LongDIR}"
+	recon_all_cmd+=" -base ${TemplateID}"
```

Comment:
Please make this indentation consistent, it bounces between a tab and 2 spaces (and the next-level lines have a tab and then 2 spaces).  Personally, I prefer 4 spaces per level, but I will tolerate tabs or fewer spaces as long as it is consistent in consecutive lines (and doesn't use both tabs and spaces in the same line).

Reply:
ok